### ISSUE TITLE
API documentation outdated

### ISSUE THREAD

**Issue by exa-mk** [it-novum/openITCOCKPIT#1027]
The API documentations at
   https://docs.it-novum.com/display/ODE
and
   https://docs.it-novum.com/display/ODD
are at least partly outdated.

E.g. the filters are not working as specified (examples: it's `filter[Hosts.id]=100` instead of `filter[Host.id]=100}` and `filter[servicename]` instead of `filter[Service.servicename]`.

**Comment by nook24** [comment]
The new interactive API documentation is available here: https://openitcockpit.io/api/

The version you linked needs to be updated...

**Comment by exa-mk** [comment]
@nook24: yeah I know this url also, but 1st) imho it's quite confusing and 2nd) it still links (as well as the official documentation part of the website) to the outdated pages

**Comment by defnull** [comment]
According to https://openitcockpit.io/api/, neither `filter[Host.id]` nor `filter[Hosts.id]` are valid parameters for filtering hosts. There is currently no documented way to get a specific host (or any other object type) by ID or UUID. Also, many APIs used by the UI are not documented at all. The linked "Complete technical documentation" is outdated and incorrect. The API is not versioned.

So, if I understand that correctly: The API is currently not considered stable, using it for automation requires reverse-engeneering and might break with any new release without notice, and there is no official support for it. Bummer :/

**Comment by itsdaveit** [comment]
Just to make sure, because this issue is quite old: Is it still the case, that its not possible to access a particular host by UUID?

**Comment by defnull** [comment]
Nothing changed. It is (and always was) possible to filter by host id, but the documentation at https://openitcockpit.io/api/ still does not reflect that. This issue is about documentation to be outdated or incomplete, not about a missing API feature.

**Comment by ibering** [comment]
@itsdaveit you can use the following request to filter by UUID

//for hosts

https://192.168.56.101/#!/hosts/browser/6200c6f9-634c-4cd6-9626-93667df2b6d0
https://192.168.56.101/hosts/byUuid/6200c6f9-634c-4cd6-9626-93667df2b6d0.json?angular=true

//... and for services

https://192.168.56.101/#!/services/browser/9520dc9a-02cf-4018-af42-9786443fd32a
https://192.168.56.101/services/byUuid/9520dc9a-02cf-4018-af42-9786443fd32a.json?angular=true

**Comment by defnull** [comment]
There are undocumented (and thus private and maybe unstable) APIs according to https://openitcockpit.io/api/

**Comment by itsdaveit** [comment]
@ibering Thanks.

https://192.168.56.101/hosts/byUuid/6200c6f9-634c-4cd6-9626-93667df2b6d0.json?angular=true
Works in general, but its missing some important information. The Response only contains the "host": key.

I don`t see the keys  "Hoststatus": {} and "ServicestatusSummary": {} which apear in https://openitcockpit.io/api/ documentation at /hosts/index.json.

How can i access those keys by Host UUID?

**Comment by nook24** [comment]
We have updated our API documentation. All API endpoints are now documented. Happy hacking
https://openitcockpit.io/api/

🥳
